app/controllers/comments_controller.rb

```diff
@@ -106,15 +134,84 @@ def show
     end
   end
 
+  def thread
+    return not_found if @comment_thread.deleted && !current_user&.privilege?('flag_curate')
+  end
+
+  def thread_rename
+    if @comment_thread.read_only? && !current_user.is_moderator
+      flash[:danger] = 'This thread has been locked.'
+      redirect_to comment_thread_path(@comment_thread.id)
+      return
+    end
+
+    @comment_thread.update title: params[:title]
+    redirect_to comment_thread_path(@comment_thread.id)
+  end
+
+  def thread_restrict
+    return not_found unless current_user.privilege? 'flag_curate'
+    return not_found if @comment_thread.read_only?
+
+    case params[:type]
+    when 'lock'
+      lu = nil
+      unless params[:duration].blank?
+        lu = params[:duration].to_i.days.from_now
+      end
+      @comment_thread.update(locked: true, locked_by: current_user, locked_until: lu)
+
+      redirect_to comment_thread_path(@comment_thread.id)
+      return
+    when 'archive'
+      @comment_thread.update(archived: true, archived_by: current_user)
+    when 'delete'
+      @comment_thread.update(deleted: true, deleted_by: current_user)
+    when 'follow'
+      ThreadFollower.create comment_thread: @comment_thread, user: current_user
```

Comment:
**question**: This combined with the first line of this method seems to mean that only users with the flag_curate privilege can follow a thread? Is that intentional?